pkg/reader/namespace/namespace.go

```diff
@@ -180,30 +207,49 @@ func GetMsgNamespaces(ns processapi.MsgNamespaces) *tetragon.Namespaces {
 		retVal.TimeForChildren = nil
 	}
 
-	return retVal
+	return retVal, nil
 }
 
-func GetHostNamespace() *tetragon.Namespaces {
-	if hostNamespace == nil {
-		hostNamespace = &tetragon.Namespaces{
-			Uts:             createHostNs("uts"),
-			Ipc:             createHostNs("ipc"),
-			Mnt:             createHostNs("mnt"),
-			Pid:             createHostNs("pid"),
-			PidForChildren:  createHostNs("pid_for_children"),
-			Net:             createHostNs("net"),
-			Time:            createHostNs("time"),
-			TimeForChildren: createHostNs("time_for_children"),
-			Cgroup:          createHostNs("cgroup"),
-			User:            createHostNs("user"),
+func InitHostNamespace() (*tetragon.Namespaces, error) {
+	if hostNamespaces != nil {
+		return hostNamespaces, nil
+	}
+
+	kernelVer, _, _ := kernels.GetKernelVersion(option.Config.KernelVersion, option.Config.ProcFS)
+	// time and time_for_children namespaces introduced in kernel 5.6
+	TimeNsSupport = (int64(kernelVer) >= kernels.KernelStringToNumeric("5.6.0"))
+
+	if TimeNsSupport == false {
+		logger.GetLogger().Infof("Kernel version %s doesn't support time namespaces", kernelVer)
+	}
+
+	for _, n := range listNamespaces {
+		ino, err := GetPidNsInode(1, n)
+		if err != nil {
+			// No support for time namespace
+			if (n == "time" || n == "time_for_children") && TimeNsSupport == false {
+				knownNamespaces[n] = &tetragon.Namespace{Inum: 0, IsHost: false}
+				continue
+			}
+			return nil, err
+		}
+		knownNamespaces[n] = &tetragon.Namespace{
+			Inum:   ino,
+			IsHost: true,
 		}
 	}
-	return hostNamespace
-}
 
-func createHostNs(ns string) *tetragon.Namespace {
-	return &tetragon.Namespace{
-		Inum:   GetPidNsInode(1, ns),
-		IsHost: true,
+	hostNamespaces = &tetragon.Namespaces{
```

Comment:
Do we really need both `hostNamespaces` and `knownNamespaces`? Can we initialize `hostNamespaces` directly?